esignet-integration-api/src/main/java/io/mosip/esignet/api/validator/AuthChallengeFactorFormatValidator.java

```
@@ -72,31 +97,61 @@ public boolean isValid(AuthChallenge authChallenge, ConstraintValidatorContext c
     }
 
     private boolean validateChallenge(String challenge) {
-        byte[] decodedBytes = Base64.getDecoder().decode(challenge);
-        String decodedString = new String(decodedBytes, StandardCharsets.UTF_8);
-        Map<String, String> challengeMap;
         try {
-            challengeMap = objectMapper.readValue(decodedString, new TypeReference<Map<String, String>>() {});
-            return fieldDetailList.stream().allMatch( fieldDetail -> isValid(fieldDetail, challengeMap) );
-        } catch (JsonProcessingException e) {
-            log.error("Failed to parse the input challenge", e);
-            return false;
-        }
-    }
+            byte[] decodedBytes = Base64.getDecoder().decode(challenge);
+            String decodedString = new String(decodedBytes, StandardCharsets.UTF_8);
+            Map<String, String> challengeMap = objectMapper.readValue(decodedString, new TypeReference<>() {});
 
-    private boolean isValid(Map<String, String> fieldDetail, Map<String, String> challengeMap) {
-        if(fieldDetail.get("type").equals("text") && !fieldDetail.get("id").equals(idField)) {
-            String value = challengeMap.get(fieldDetail.get("id"));
-            if(!StringUtils.hasText(value))
+            JsonNode schemaArray = fieldJson.path("schema");
+            if (!schemaArray.isArray()) {
+                log.warn("Schema is not an array");
```

If the schema field is not an array then it should throw an error when we load the schema not while validation of dto.

---

updated @sacrana0